doc/riscv_trace.rst

```diff
@@ -0,0 +1,70 @@
+.. _riscv_trace:                                                                                         
+
+
+RISC-V Trace Interface
+============================
+
+Ibex implements an instruction trace interface compatible with the `RISC-V Trace Specification, version 1.0, as released on March 20, 2020 <https://github.com/riscv/riscv-trace-spec/>`_. 
+This interface exports needed internal signals as ibex_core outputs to be used to encode the trace packets to be logged.
+
+Signals implemented until now are shown in the table below:
+
+
+Trace Signals
+-------------
+
++--------------------+---------------------+-------------------------------------------------------------+
+|   Signal Name      |   Number of bits    |                         Description                         |
++====================+=====================+=============================================================+
+| rv_trace_ctype     |           2         | 0: Don’t report;                                            |
+|                    |                     | 1: Report imprecisely;                                      |
+|                    |                     | 2: Report precisely;                                        |
+|                    |                     | 3: Report as asynchronous discontinuity.                    |
++--------------------+---------------------+-------------------------------------------------------------+
+| rv_trace_itype     |           3         | 0: Final instruction in block is none of other itype codes; |
+|                    |                     | 1: Exception. An exception that traps occurred              |
+|                    |                     | following the final retired instruction in the block;       |
+|                    |                     | 2: Interrupt. An interrupt that traps occurred              |
+|                    |                     | following the final retired instruction in the block;       |
+|                    |                     | 3: Exception or interrupt return;                           |
+|                    |                     | 4: Nontaken branch;                                         |
+|                    |                     | 5: Taken branch;                                            |
+|                    |                     | 6: Uninferable jump if itype_width_p is 3                   |
++--------------------+---------------------+-------------------------------------------------------------+
+| rv_trace_iaddr     |          32         | Contains the value of the address in the cases it needs     |
```

Comment:
iaddr is *always* reported, there's no trace filtering happened yet.

Reply:
So you mean we will be selecting when to actually to report it later by the encoder, and here we just describe the interface in ibex-core?

Reply:
Yes, the interface you're describing here is the core interface; it's effectively the text form of the signals you're creating in `ibex_core.sv`.